src/components/Experience.ts

```diff
@@ -25,15 +40,103 @@ export class Experience<TGameStartr extends FullScreenPokemon> extends Component
     }
 
     /**
-     * Gives experience to a specified pokemon
+     * Gives experience to a specified pokemon.
      *
      * @param pokemon   The pokemon to give experience to.
      * @param experience   The amount of experience to give.
      */
     public gainExperience(pokemon: IPokemon, experience: number): void {
         pokemon.experience += experience;
-        if (pokemon.experience >= this.gameStarter.equations.experienceStarting(pokemon.title, pokemon.level + 1)) {
+        if (pokemon.experience >= this.gameStarter.equations.experienceStarting(pokemon.title, pokemon.level + 1)
+            && pokemon.statistics.health.current !== 0) {
             this.levelup(pokemon);
         }
     }
+
+    /**
+     * Calculates how much experience a defeated Pokemon is worth.
+     * https://bulbapedia.bulbagarden.net/wiki/Experience#Gain_formula
+     *
+     * @param pokemon   The defeated Pokemon.
+     * @param trainer   If the defeated Pokemon had a trainer.
+     */
+    public calculateExperience(pokemon: IPokemon, trainer: boolean): IExperienceGains[] {
+        const participatedPokemon: IPokemon[] = this.gameStarter.itemsHolder.getItem("battleParticipants");
+        const numerator: number = this.calculateNumerator(pokemon.level, trainer,
```

Comment:
Same alignment complaint as the other place (move the first arg to the next line).